lib/testers/rule-tester.js

```diff
@@ -296,9 +285,11 @@ RuleTester.prototype = {
             } else {
                 code = item.code;
 
-                // Assumes everything on the item is a config except for the
-                // parameters used by this tester
-                const itemConfig = lodash.omit(item, RuleTesterParameters);
+                // Validate only config properties
+                const schemaProperties = Object.keys(configSchema.properties);
+
+                // const schemaValidator = validator(configSchema, { verbose: true });
+                const itemConfig = lodash.pick(item, schemaProperties);
```

Comment:
I definitely think it would be worthwhile. It turns out that only one rule actually does this intentionally, and the others just have typos (see https://github.com/eslint/eslint/pull/8305). Applying [this diff](https://gist.github.com/not-an-aardvark/d879e6273318fd85be2694502c3f3dad) to your current branch would fix the one rule that does it intentionally and add the validation back.